src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -63,6 +63,51 @@ static void SendUprobesFullAddressInfoToListener(
   listener->OnAddressInfo(std::move(address_info));
 }
 
+static bool CallstackIsInUserSpaceInstrumentation(
+    const std::vector<unwindstack::FrameData>& frames,
+    const UserSpaceInstrumentationAddresses& user_space_instrumentation_addresses) {
+  CHECK(!frames.empty());
+
+  // This case is for a sample falling directly inside a user space instrumentation trampoline.
+  if (user_space_instrumentation_addresses.IsInEntryTrampoline(frames.front().pc) ||
+      user_space_instrumentation_addresses.IsInReturnTrampoline(frames.front().pc)) {
+    return true;
+  }
+
+  // This case is for all samples falling in a callee of the trampoline. These are normally in the
+  // injected library, but they could also be in a module containing a function called by the
+  // library, hence why we look at all frames for that library.
+  std::string_view injected_library_map_name =
+      user_space_instrumentation_addresses.GetInjectedLibraryMapName();
+  return std::any_of(frames.begin(), frames.end(),
```

Comment:
So this ich checking that no frame is in "liborbituserspaceinstrumentation.so", right?

Reply:
Exactly. I rephrased the comment a tiny bit.